ladybug/_datacollectionbase.py

```diff
@@ -940,6 +940,18 @@ def is_mutable(self):
         """Boolean denoting whether the data collection is mutable."""
         return self._mutable
 
+    def __key(self):
+        return (
+            self.header, self.values, self.datetimes, self.validated_a_period,
+            self._enumeration
+        )
+
+    def __eq__(self, other):
+        return isinstance(other, BaseCollection) and self.__key() == other.__key()
```

Comment:
Unless you want to re-write this function each time it appears in the 8 classes that inherit from it, I would change this to:
```
return isinstance(other, self.__class__) and self.__key() == other.__key()
```
As it is right now, it could produce some odd results.